routes/signup.js

```diff
@@ -0,0 +1,31 @@
+const express = require('express');
+const bcrypt = require('bcrypt');
+const User = require('./../models/userModel.js');
+const jwt = require('jsonwebtoken');
+
+const router = express.Router();
+
+const generateToken = (Object) => {
+  const signature = 'my-super-secret';
+  const expiration = '6h';
+  return jwt.sign({ Object }, signature, { expiresIn: expiration });
+};
+
+router.post('/', async (req, res) => {
+  const newUser = req.body;
+  const token = generateToken(newUser);

```

Comment:
You don't need to pass all fields. Only use ID will suffice. Also, do not generate token BEFORE creating use in the database.